### PR TITLE
profile page

### PR DESCRIPTION
**Pull request by M-Rb3**
image
image

### CODE REVIEW

**Comment by lachlanglen** [comment]
@M-Rb3 could you summarize what has been implemented in this PR? Thanks

**Comment by M-Rb3** [comment]
- add profile page, so donors can see their donations.
- the profile page contains the following tabs(feed, donations), and followers but it is hidden you can redirected to there after clicking the followers Stats on the profile page.
- the donations table includes direct donations and sponsorships, and it is disabled for project page for now, till I am done with it.

**Comment by M-Rb3** [comment]
- Moved some components from the project dir to profile that will be shared for both pages, so we can easily update the UI later.  The logic is done on the (-.Detail). And the UI on the profile.body

File: apps/potlock/widget/Project/Detail.jsx [hunk 1]
```diff
@@ -1,235 +1,127 @@
...
-            This project status is {project.status} and has not been approved.
-          </BannerText>
-        </Row>
-        {project.review_notes && (
-          <BannerText style={{ fontStyle: "italic" }}>
-            Admin review notes: {project.review_notes}
-          </BannerText>
-        )}
-      </Banner>
-    )}
```
Comment (lachlanglen) [inline apps/potlock/widget/Project/Detail.jsx:167]: I see this was broken out to its own component (`Project.ProjectBanner`) but this component is never used, so the banner is no longer there. Can you use this widget here so that the banner still displays, thanks.

File: apps/potlock/widget/Project/ProjectBanner.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      </BannerAlertSvg>
+      <BannerText>This project status is {project.status} and has not been approved.</BannerText>
+    </Row>
+    {project.review_notes && (
+      <BannerText style={{ fontStyle: "italic" }}>
+        Admin review notes: {project.review_notes}
+      </BannerText>
+    )}
+  </Banner>
+);
```
Comment (lachlanglen) [inline apps/potlock/widget/Project/ProjectBanner.jsx:51]: as noted above, this was broken out but this widget is never used, it should be used in `Project.Detail`

**Review by lachlanglen — CHANGES_REQUESTED** [review]
Hey, thanks for all the great work here @M-Rb3. Please check out my comments and address when you can.

In the future, please try to avoid addressing multiple features/fixes in a single PR, as it 1) makes the PR extra large, increasing likelihood of new bugs introduced, and 2) makes the review process take longer.

Thanks! I appreciate your work! 🫡

File: apps/potlock/widget/Profile/BodyHeader.jsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      permissions.includes(`${kind}:${action}`) ||
+      permissions.includes(`${kind}:*`) ||
+      permissions.includes(`*:${action}`) ||
+      permissions.includes("*:*")
+    );
+  });
+  return allowed;
+}, [policy]);
+
+const isOwner = props.projectId === context.accountId;
```
[the remark this answers is not in the document]
Comment (M-Rb3) [reply apps/potlock/widget/Profile/BodyHeader.jsx:95]: I did not update this part, it was in the previous code on Poject.Body. but from what I see it checks if the signed-in address is the same as the project, if yes, it enables editing the project

image

File: apps/potlock/widget/Project/Options.jsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    id: "following",
+    disabled: false,
+    source: `${props.ownerId}/widget/Profile.FollowTabs`,
+  },
+];
+
+return {
+  ProjectOptions,
+  ProfileOptions,
+};
```
[the remark this answers is not in the document]
Comment (M-Rb3) [reply apps/potlock/widget/Project/Options.jsx:84]: done

File: apps/potlock/widget/Profile/DonationsTable.jsx [hunk 5]
```diff
@@ -0,0 +1,331 @@
...
+            setPage(page);
+            setTotalDonation(donations.slice(page * perPage, (page + 1) * perPage));
+            setFilteredDonations(donations.slice(page * perPage, (page + 1) * perPage));
+          },
+          data: donations,
+          page: page,
+          perPage: perPage,
+          bgColor: "#7B7B7B",
+        }}
+      />
```
[the remark this answers is not in the document]
Comment (M-Rb3) [reply apps/potlock/widget/Profile/DonationsTable.jsx:328]: I will update it

**Review by lachlanglen — APPROVED** [review]
